core/src/main/java/org/mskcc/cbio/portal/dao/JdbcUtil.java

```diff
@@ -69,6 +69,28 @@ public static void setDataSource(DataSource value) {
         dataSource = value;
     }
 
+    private static DataSource initDataSource() {
+        DatabaseProperties dbProperties = DatabaseProperties.getInstance();
+        String host = dbProperties.getDbHost();
+        String userName = dbProperties.getDbUser();
+        String password = dbProperties.getDbPassword();
+        String database = dbProperties.getDbName();
+        String url ="jdbc:mysql://" + host + "/" + database +
```

Comment:
The overriding of userName, password, and other connection string terms is also being handled more subtly than may first appear here. I think this change also needs careful testing to make sure that properties passed in either through the portal.properties files or through java "-Dproperty_name=property_value" clauses will still work correctly.

Reply:
will revert the change.